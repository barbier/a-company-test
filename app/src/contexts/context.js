import { createContext } from 'react'

export const ArticleContext = createContext()
