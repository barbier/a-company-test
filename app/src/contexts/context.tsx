import React, { useState, createContext, useContext } from "react";
import { IArticle } from "../interfaces/Article";
import { IContext } from "../interfaces/ContextProvider";

const useArticle = (): any => {
  const [selectedArticle, setSelectedArticle] = useState({});

  return {
    selectedArticle,
    chooseArticle: (article: IArticle) => setSelectedArticle(article),
  };
};

const ArticleContext = createContext({} as any);

export const ArticleContextProvider: React.FC<{ children: any }> = ({
  children,
}: IContext) => (
  <ArticleContext.Provider value={useArticle()}>
    {children}
  </ArticleContext.Provider>
);

export const useSelectedArticle = (): any =>
  useContext(ArticleContext).selectedArticle;
export const useSetSelectedArticle = (article: IArticle): any =>
  useContext(ArticleContext).chooseArticle(article);
