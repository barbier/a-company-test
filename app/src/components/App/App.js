import { useEffect, useState } from 'react'
import ArticleContentWrapper from '../articlecontent/ArticleContent'
import { AsideContent, Footer, FooterText, Header, HeaderTitle, Main } from '../../styles/styles'
import ArticlesList from '../articleslist/ArticlesList'
import { ArticleContext } from '../../contexts/context'

function App() {
  const [articles, setArticles] = useState([])
  const [selectedArticle, setSelectedArticle] = useState({})
  
  useEffect(() => {
    const fetchData = async () => {
      const result = await fetch('http://localhost:8000/v1/news?q=')
      const resultJson = await result.json()
      setArticles(resultJson.articles)
    }

    fetchData()
  }, [])

  return (
    <ArticleContext.Provider value={[selectedArticle, setSelectedArticle]}>
      <Header>
        <HeaderTitle>
          Newspaper
        </HeaderTitle>
      </Header>
      <Main>
        <AsideContent>
          <ArticlesList articles={articles} />
        </AsideContent>
        <ArticleContentWrapper />
      </Main>
      <Footer>
        <FooterText>
          Created by: Barbier
        </FooterText>
      </Footer>
    </ArticleContext.Provider>
  )
}

export default App
