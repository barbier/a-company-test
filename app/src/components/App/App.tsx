import React, { useEffect, useState } from "react";
import ArticleContentWrapper from "../articlecontent/ArticleContent";
import {
  AsideContent,
  Footer,
  FooterText,
  Header,
  HeaderTitle,
  Main,
} from "../../styles/styles";
import ArticlesList from "../articleslist/ArticlesList";
import { ArticleContextProvider } from "../../contexts/context";

const App: React.FC = () => {
  const [articles, setArticles] = useState([]);

  useEffect(() => {
    const fetchData = async (): Promise<void> => {
      const result = await fetch("http://localhost:8000/v1/news?q=");
      const resultJson = await result.json();
      setArticles(resultJson.articles);
    };

    void fetchData();
  }, []);

  return (
    <ArticleContextProvider>
      <Header>
        <HeaderTitle>Newspaper</HeaderTitle>
      </Header>
      <Main>
        <AsideContent>
          <ArticlesList articles={articles} />
        </AsideContent>
        <ArticleContentWrapper />
      </Main>
      <Footer>
        <FooterText>Created by: Barbier</FooterText>
      </Footer>
    </ArticleContextProvider>
  );
};

export default App;
