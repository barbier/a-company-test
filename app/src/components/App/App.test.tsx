import renderer from 'react-test-renderer'
import App from './App'

import { ArticleContextProvider } from "../../contexts/context";

describe("Test Articles List rendering options", () => {
  test("It should match snapshot", () => {
    const component = renderer.create(
      <ArticleContextProvider>
        <App />
      </ArticleContextProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
// test('renders learn react link', () => {
//   render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
