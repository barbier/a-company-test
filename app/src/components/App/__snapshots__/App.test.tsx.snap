// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Articles List rendering options It should match snapshot 1`] = `
Array [
  <header
    className="css-1gw4kos"
  >
    <h1
      className="css-1ilhfkm"
    >
      Newspaper
    </h1>
  </header>,
  <main
    className="css-8u1dut"
  >
    <aside
      className="css-6p15e3"
    >
      <p>
        Failed to fetch data
      </p>
    </aside>
    <article
      className="css-6p15e3"
    >
      <header
        className="css-rg0tw2"
      >
        <h1
          className="css-q28qg0"
        />
        <h2
          className="css-q28qg0"
        >
           Author: 
          Unknown
           
        </h2>
        <small
          className="css-0"
        >
          Published at: 
          Invalid Date
        </small>
      </header>
      <img
        className="css-1t0xj7x"
      />
      <p
        className="css-1k46fjq"
      />
      <footer
        className="css-0"
      >
        Source:
        <a
          className="css-hnw5at"
          target="_blank"
        />
      </footer>
    </article>
  </main>,
  <footer
    className="css-1npnmb4"
  >
    <p
      className="css-0"
    >
      Created by: Barbier
    </p>
  </footer>,
]
`;
